File: Backend/Travle.Model/Requests/BookingInsertRequest.cs
namespace Travle.Model.Requests
{
    /// <summary>
    /// A traveler's request to book a tour schedule. The user is taken from the JWT (never the client)
    /// and the total is computed server-side from the tour's price — only the slot and the group size
    /// come from the client. Creation enters the booking as PaymentInProgress and holds capacity for
    /// 15 minutes while payment completes.
    /// </summary>
    public class BookingInsertRequest
    {
        public int TourScheduleId { get; set; }

        /// <summary>Number of seats to reserve (checked against the slot's live free capacity server-side).</summary>
        public int NumberOfPeople { get; set; }
    }
}

File: UI/travle_core/lib/src/models/booking_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'booking_insert_request.g.dart';

/// A traveler's request to book a tour schedule (mirrors the backend
/// `BookingInsertRequest`). The user and the total are resolved server-side —
/// only the slot and the group size come from the client.
@JsonSerializable()
class BookingInsertRequest {
  BookingInsertRequest({
    required this.tourScheduleId,
    required this.numberOfPeople,
  });

  final int tourScheduleId;
  final int numberOfPeople;

  factory BookingInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$BookingInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$BookingInsertRequestToJson(this);
}
